spi/src/test/java/com/amazon/opendistroforelasticsearch/jobscheduler/spi/utils/LockServiceIT.java

```diff
@@ -86,178 +89,343 @@
+                                        if (lock != null) {
+                                            lockModelAtomicReference.set(lock);
+                                            multiThreadCreateLockCounter.getAndAdd(1);
+                                        }
+                                        callableLatch.countDown();
+                                    },
+                                    exception -> fail(exception.getMessage())
+                            ));
+                            callableLatch.await(10000L, TimeUnit.MILLISECONDS);
+                            return true;
+                        };
+
+                        List<Callable<Boolean>> callables = Arrays.asList(
+                                callable,
+                                callable,
+                                callable
+                        );
+
+                        executor.invokeAll(callables)
+                                .forEach(future -> {
+                                    try {
+                                        future.get();
+                                    } catch (Exception e) {
+                                        fail(e.getMessage());
+                                    }
+                                });
+                        executor.shutdown();
+                        executor.awaitTermination(10000L, TimeUnit.MILLISECONDS);
+
+                        assertEquals("There should be only one that grabs the lock.", 1, multiThreadCreateLockCounter.get());
+
+                        final LockModel lock = lockModelAtomicReference.get();
+                        assertNotNull("Expected to successfully grab lock", lock);
+                        lockService.release(lock, ActionListener.wrap(
+                                released -> {
+                                    assertTrue("Failed to release lock.", released);
+                                    lockService.deleteLock(lock.getLockId(), ActionListener.wrap(
+                                            deleted -> {
+                                                assertTrue("Failed to delete lock.", deleted);
+                                                latch.countDown();
+                                            },
+                                            exception -> fail(exception.getMessage())
+                                    ));
+                                },
+                                exception -> fail(exception.getMessage())
+                        ));
+                    } else {
+                        fail("Failed to create lock index.");
+                    }
+                },
+                exception -> fail(exception.getMessage())
+        ));
+        assertTrue("Test timed out - possibly leaked into other tests", latch.await(30000L, TimeUnit.MILLISECONDS));
     }
 
+    private volatile static AtomicInteger multiThreadAcquireLockCounter = new AtomicInteger(0);
```

Comment:
Does this variable need to be outside of the `testMultiThreadAcquireLock` test function?

Reply:
Should use volatile, otherwise it may fetch the value from CPU cache, which could be inaccurate since another thread may have already updated the value